hw8/todolist/src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from "react";
+import TextInput from "../TextInput/TextInput";
+import Button from "../Button/Button";
+import DateInput from "../DateInput/DateInput";
+
+import "./Modal.css";
+import { Task } from "../../interfaces/Task";
+import { addTask } from "../../taskSlice";
+import { useDispatch } from "react-redux";
+
+interface Props {
+  modalVisibility: boolean;
+  toggleModalVisibility: () => void;
+}
+
+const Modal = ({ modalVisibility, toggleModalVisibility }: Props) => {
+  const [newTaskName, setNewTaskName] = useState("");
+  const [newDeadline, setNewDeadline] = useState(new Date());
+  const [newTag, setNewTag] = useState("tag");
+
+  const dispatch = useDispatch();
+
+  const onAddTask = async (newTask: Task) => {
+    const response = await fetch("http://localhost:3004/tasks", {
+      method: "POST",
```

Comment:
Don't store requests in the components. Extract it to the service method. Also, you can also create an action for it with createAsyncThunk and use it with the new service method, so the component will just work with the prepared functions and data